pkg/controller/workload/reconciler.go

```diff
@@ -313,7 +313,7 @@ func (r *Reconciler) cleanupOrphanedObjects(ctx context.Context, previousResourc
 		}
 		orphaned := true
 		for _, realizedResource := range realizedResources {
-			if realizedResource.StampedRef.GroupVersionKind() == prevResource.StampedRef.GroupVersionKind() &&
+			if realizedResource.StampedRef != nil && realizedResource.StampedRef.GroupVersionKind() == prevResource.StampedRef.GroupVersionKind() &&
```

Comment:
wdyt of instead breaking out of the loop preemptively?

ala

```
for _, realizedResource := range realizedResources {
  if realizedResource.StampedRef == nil {
     continue
  }
}
```

Reply:
Done. And ported to deliverable 😅